UnhollowerBaseLib/Il2CppObjectBase.cs

```diff
@@ -66,7 +66,7 @@ public Il2CppObjectBase(IntPtr pointer)
             if (!IL2CPP.il2cpp_class_is_assignable_from(nestedTypeClassPointer, ownClass))
                 return null;
 
-            if (RuntimeSpecificsStore.IsInjected(ownClass))
+            if (ownClass == nestedTypeClassPointer && RuntimeSpecificsStore.IsInjected(ownClass))
```

Comment:
What is the rationale behind exact-class check? This can cause a new instance of an injected class to be created, which would result in TryCast returning clean wrappers with unset fields if a wrong (but compatible) type was specified.

Reply:
The idea was to prevent it from trying to do invalid casts to interfaces that aren't known to the proxy assemblies.
I guess this could break if you try to cast from one injected class that extends another injected class to the latter.